spec_integration/queue_runner.rb

```diff
@@ -0,0 +1,75 @@
+require 'knapsack_pro'
+require 'json'
+
+ENV['KNAPSACK_PRO_CI_NODE_BUILD_ID'] = SecureRandom.uuid
+ENV['KNAPSACK_PRO_TEST_SUITE_TOKEN_RSPEC'] = 'fake-api-token'
+ENV['KNAPSACK_PRO_TEST_DIR'] = 'spec_integration'
+ENV['KNAPSACK_PRO_TEST_FILE_PATTERN'] = "spec_integration/**{,/*/**}/*_spec.rb"
+
+RSPEC_OPTIONS = ENV.fetch('TEST__RSPEC_OPTIONS')
+SHOW_DEBUG_LOG = ENV['TEST__SHOW_DEBUG_LOG'] == 'true'
+BATCHED_TESTS = JSON.load(ENV.fetch('TEST__BATCHED_TESTS'))
+
+module KnapsackProExtensions
+  module QueueAllocatorExtension
+    def test_file_paths(can_initialize_queue, executed_test_files)
+      @@batch_index ||= 0
+      batches = BATCHED_TESTS + [
+        [], # the last Queue API response is always an empty list of test files
+      ]
+      tests = batches[@@batch_index] || []
+      @@batch_index += 1
+
+      if SHOW_DEBUG_LOG
+        puts "DEBUG: mocked tests from the Queue API: #{tests.inspect}"
+      end
+
+      return tests
+    end
+  end
+
+  module Report
+    def create_build_subset(test_files)
+      if ENV['TEST__LOG_EXECUTION_TIMES']
+        have_execution_time = test_files.all? { _1.fetch('time_execution') > 0 }
+        puts "LOG_EXECUTION_TIMES: test_files: #{test_files.size}, test files have execution time: #{have_execution_time}"
```

Comment:
All the `puts` in this file are mixed with the output of Knapsack. Should we add a prefix to all of them so that it's less confusing if a spec fails? Something like `[INTEGRATION TEST] ...`.

Reply:
Good idea. I applied it.